[Bug] API 28: Can't open Filtered Deck Options

* make a custom study deck
* Long press - Deck Options

`android.content.res.Resources$NotFoundException: File res/color/faded_primary.xml from drawable resource ID #0x7f06007d`

`Binary XML file line #17: <item> tag requires a 'drawable' attribute or child tag defining a drawable`

```
ACRA caught a RuntimeException for com.ichi2.anki.debug
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki.debug/com.ichi2.anki.FilteredDeckOptions}: android.view.InflateException: Binary XML file line #27: Binary XML file line #27: Error inflating class androidx.appcompat.widget.ActionBarContainer
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6077)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
Caused by: android.view.InflateException: Binary XML file line #27: Binary XML file line #27: Error inflating class androidx.appcompat.widget.ActionBarContainer
Caused by: android.view.InflateException: Binary XML file line #27: Error inflating class androidx.appcompat.widget.ActionBarContainer
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
	at android.view.LayoutInflater.createView(LayoutInflater.java:645)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:787)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
	at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:979)
	at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:889)
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:772)
	at com.ichi2.ui.AppCompatPreferenceActivity.setContentView(AppCompatPreferenceActivity.kt:211)
	at android.preference.PreferenceActivity.onCreate(PreferenceActivity.java:557)
	at com.ichi2.ui.AppCompatPreferenceActivity.onCreate(AppCompatPreferenceActivity.kt:185)
	at com.ichi2.anki.FilteredDeckOptions.onCreate(FilteredDeckOptions.kt:200)
	at android.app.Activity.performCreate(Activity.java:6662)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2599)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6077)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
Caused by: android.content.res.Resources$NotFoundException: Drawable com.ichi2.anki.debug:color/faded_primary with resource ID #0x7f06007d
Caused by: android.content.res.Resources$NotFoundException: File res/color/faded_primary.xml from drawable resource ID #0x7f06007d
	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:725)
	at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:571)
	at android.content.res.Resources.loadDrawable(Resources.java:854)
	at android.content.res.TypedArray.getDrawable(TypedArray.java:928)
	at androidx.appcompat.widget.ActionBarContainer.<init>(ActionBarContainer.java:68)
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
	at android.view.LayoutInflater.createView(LayoutInflater.java:645)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:787)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
	at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:979)
	at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:889)
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:772)
	at com.ichi2.ui.AppCompatPreferenceActivity.setContentView(AppCompatPreferenceActivity.kt:211)
	at android.preference.PreferenceActivity.onCreate(PreferenceActivity.java:557)
	at com.ichi2.ui.AppCompatPreferenceActivity.onCreate(AppCompatPreferenceActivity.kt:185)
	at com.ichi2.anki.FilteredDeckOptions.onCreate(FilteredDeckOptions.kt:200)
	at android.app.Activity.performCreate(Activity.java:6662)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2599)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6077)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #17: <item> tag requires a 'drawable' attribute or child tag defining a drawable
	at android.graphics.drawable.StateListDrawable.inflateChildElements(StateListDrawable.java:183)
	at android.graphics.drawable.StateListDrawable.inflate(StateListDrawable.java:116)
	at android.graphics.drawable.DrawableInflater.inflateFromXml(DrawableInflater.java:130)
	at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:1227)
	at android.graphics.drawable.Drawable.createFromXml(Drawable.java:1200)
	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:715)
	... 33 more
```

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/FilteredDeckOptionsTest.kt
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/FilteredDeckOptionsTest.kt
@@ -1,0 +1,46 @@
+/*
+ *  Copyright (c) 2024 David Allison <davidallisongithub@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki
+
+import androidx.core.content.edit
+import androidx.test.core.app.ActivityScenario
+import com.ichi2.anki.tests.InstrumentedTest
+import com.ichi2.anki.testutil.GrantStoragePermission
+import com.ichi2.anki.testutil.grantPermissions
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class FilteredDeckOptionsTest : InstrumentedTest() {
+    @get:Rule
+    val runtimePermissionRule = grantPermissions(GrantStoragePermission.storagePermission)
+
+    @Before
+    fun before() {
+        AnkiDroidApp.sharedPrefs().edit {
+            putBoolean(IntroductionActivity.INTRODUCTION_SLIDES_SHOWN, true)
+        }
+    }
+
+    @Test
+    fun canOpenDeckOptions() {
+        ActivityScenario.launch(DeckPicker::class.java).onActivity { deckPicker ->
+            val deckId = col.decks.newDyn("Filtered Testing")
+            deckPicker.showContextMenuDeckOptions(deckId)
+        }.close()
+    }
+}

--- a/AnkiDroid/src/main/res/values-v29/theme_dark.xml
+++ b/AnkiDroid/src/main/res/values-v29/theme_dark.xml
@@ -1,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~  Copyright (c) 2024 David Allison <davidallisongithub@gmail.com>
+  ~
+  ~  This program is free software; you can redistribute it and/or modify it under
+  ~  the terms of the GNU General Public License as published by the Free Software
+  ~  Foundation; either version 3 of the License, or (at your option) any later
+  ~  version.
+  ~
+  ~  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+  ~  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+  ~  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+  ~
+  ~  You should have received a copy of the GNU General Public License along with
+  ~  this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
+    <style name="Base.V29.Dark" parent="@style/Base.Theme.Dark">
+        <item name="colorSurfaceContainer">@color/faded_primary</item>
+    </style>
+
+    <style name="Theme_Dark" parent="Base.V29.Dark"/>
+</resources>

--- a/AnkiDroid/src/main/res/values-v29/theme_light.xml
+++ b/AnkiDroid/src/main/res/values-v29/theme_light.xml
@@ -1,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~  Copyright (c) 2024 David Allison <davidallisongithub@gmail.com>
+  ~
+  ~  This program is free software; you can redistribute it and/or modify it under
+  ~  the terms of the GNU General Public License as published by the Free Software
+  ~  Foundation; either version 3 of the License, or (at your option) any later
+  ~  version.
+  ~
+  ~  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+  ~  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+  ~  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+  ~
+  ~  You should have received a copy of the GNU General Public License along with
+  ~  this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
+    <style name="Base.V29.Light" parent="@style/Base.Theme.Light">
+        <item name="colorSurfaceContainer">@color/faded_primary</item>
+    </style>
+
+    <style name="Theme_Light" parent="Base.V29.Light"/>
+</resources>

--- a/AnkiDroid/src/main/res/values-v29/theme_plain.xml
+++ b/AnkiDroid/src/main/res/values-v29/theme_plain.xml
@@ -1,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~  Copyright (c) 2024 David Allison <davidallisongithub@gmail.com>
+  ~
+  ~  This program is free software; you can redistribute it and/or modify it under
+  ~  the terms of the GNU General Public License as published by the Free Software
+  ~  Foundation; either version 3 of the License, or (at your option) any later
+  ~  version.
+  ~
+  ~  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+  ~  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+  ~  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+  ~
+  ~  You should have received a copy of the GNU General Public License along with
+  ~  this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
+    <style name="Base.V29.Light.Plain" parent="@style/Base.Theme.Light.Plain">
+        <item name="colorSurfaceContainer">@color/faded_primary</item>
+    </style>
+
+    <style name="Theme_Light.Plain" parent="Base.V29.Light.Plain"/>
+</resources>

--- a/AnkiDroid/src/main/res/values/theme_dark.xml
+++ b/AnkiDroid/src/main/res/values/theme_dark.xml
@@ -9,7 +9,7 @@
     <color name="theme_dark_drawer_row_current">#33999999</color>
     <color name="theme_dark_drawer_item">#DEFFFFFF</color>
 
-    <style name="Theme_Dark" parent="@style/Theme.Material3.Dark.NoActionBar">
+    <style name="Base.Theme.Dark" parent="@style/Theme.Material3.Dark.NoActionBar">
         <!-- Android colors -->
         <item name="colorPrimary">@color/material_blue_400</item>
         <item name="colorOnPrimary">@color/white</item>
@@ -27,7 +27,7 @@
 
         <item name="colorSurfaceContainerHighest">?attr/colorSurfaceContainer</item>
         <item name="colorSurfaceContainerHigh">?android:attr/colorBackground</item>
-        <item name="colorSurfaceContainer">@color/faded_primary</item>
+        <item name="colorSurfaceContainer">#0F42A5F5</item>
         <item name="colorSurfaceContainerLow">?android:attr/colorBackground</item>
         <!-- Widget colors -->
         <item name="widgetColorActivated">?attr/colorAccent</item>
@@ -155,4 +155,6 @@
         <item name="alertDialogTheme">@style/AlertDialogStyle</item>
 
     </style>
+
+    <style name="Theme_Dark" parent="Base.Theme.Dark"/>
 </resources>

--- a/AnkiDroid/src/main/res/values/theme_light.xml
+++ b/AnkiDroid/src/main/res/values/theme_light.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 
 <resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
-    <style name="Theme_Light" parent="@style/Theme.Material3.Light.NoActionBar">
+    <style name="Base.Theme.Light" parent="@style/Theme.Material3.Light.NoActionBar">
         <!-- Android colors -->
         <item name="colorPrimary">@color/material_light_blue_500</item>
         <item name="colorPrimaryDark">@color/material_light_blue_700</item>
@@ -13,7 +13,7 @@
 
         <item name="colorSurfaceContainerHighest">?attr/colorSurfaceContainer</item>
         <item name="colorSurfaceContainerHigh">?android:attr/colorBackground</item>
-        <item name="colorSurfaceContainer">@color/faded_primary</item>
+        <item name="colorSurfaceContainer">#0F03A9F4</item>
         <item name="colorSurfaceContainerLow">?android:attr/colorBackground</item>
 
         <item name="android:textColor">@color/text_color_light</item>
@@ -143,4 +143,6 @@
         <item name="progressDialogButtonTextColor">@color/material_light_blue_500</item>
 
     </style>
+
+    <style name="Theme_Light" parent="Base.Theme.Light"/>
 </resources>

--- a/AnkiDroid/src/main/res/values/theme_plain.xml
+++ b/AnkiDroid/src/main/res/values/theme_plain.xml
@@ -6,11 +6,12 @@
     <color name="theme_plain_accent">#607D8B</color>
     <color name="theme_plain_primary_text">#212121</color>
 
-    <style name="Theme_Light.Plain">
+    <style name="Base.Theme.Light.Plain">
         <!-- Android colors -->
         <item name="colorPrimary">@color/theme_plain_primary</item>
         <item name="colorPrimaryDark">@color/theme_plain_primary_dark</item>
         <item name="colorAccent">@color/theme_plain_accent</item>
+        <item name="colorSurfaceContainer">#0F9E9E9E</item>
         <item name="android:textColor">@color/text_color_light</item>
         <item name="android:textColorSecondary">@color/text_color_secondary_light</item>
         <item name="preferenceCategoryTitleTextColor">?attr/colorAccent</item>
@@ -72,4 +73,6 @@
         <item name="progressDialogButtonTextColor">@color/theme_plain_primary</item>
 
     </style>
+
+    <style name="Theme_Light.Plain" parent="Base.Theme.Light.Plain"/>
 </resources>

